### PR TITLE
Oakland template

### PR DESCRIPTION
**Pull request by dimkarakostas**

### CODE REVIEW

File: etc/theory/compression_security.tex [hunk 1]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+below.
+
+Firstly, he should be able to issue any number of malicious requests using the
+user's cookies. This can be achieved by forcing the user's browser to run a
+piece of code, such as a Javascript script hosted on an adversary-controlled
+website, that requests some web page on said website.
+
+Secondly, the requests should include an adversary-chosen string that is also
+included in the response data. In order to achieve that, the website must offer
+web page that provides a HTTP GET parameter whose value is included in the HTML
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: **some** web page

File: etc/theory/compression_security.tex [hunk 2]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+
+Finally, the adversary is able to handle all user's network traffic. That way he
+is able to collect and analyze the network packets of the encrypted
+communication with the server. He is also able inject the malicious script in
+plain HTTP responses the user receives from third websites even when the user is
+careful enough to avoid adversary-controlled websites.
+
+The described adversary can issue a compression side-channel attack against the
+user and the website. The malicious requests contain specially crafted
+reflection strings that is included in the response and compressed with the
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: are included

File: etc/theory/compression_security.tex [hunk 3]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+private data. The structure of the reflection depends heavily on properties of
+the private data and will be further explored in following sections.
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: the following sections

File: etc/theory/compression_security.tex [hunk 4]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+
+As the attack progresses the adversary collects network data for multiple
+requests and reflections. Although this data is encrypted, analysis on the
+lengths of different request packets may reveal properties of the private data,
+at which point the privacy of communication between the user and the website is
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: the communication

File: etc/theory/compression_security.tex [hunk 5]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+attacker can bypass it. Strings like "div", "ul", "Your inbox:" are considered
+static in our example.
+
+A \textit{reflection} $r$ is a component crafted by the attacker and can be
+adaptively transformed as the attack progresses. The reflection is included in
+the response and compressed in the same context with the secret that is
+targeted. In our example, the HTTP GET value "attack" of parameter "query" is used as
+a reflection.
+
+Secrets and reflections are chosen from distribution $\mathcal{M}$. For example,
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: the distribution

File: etc/theory/compression_security.tex [hunk 6]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+consider the secret "Bank" in our example. The distribution $\mathcal{M}$ may
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: Maybe use the routing number as a better example for a secret alphabet here?

File: etc/theory/compression_security.tex [hunk 7]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+describe all permutations of 4-length strings consisting of printable characters
+or constrained in only english alphabet letters. The reflection is chosen from
+the distribution $\mathcal{M}$, so the definition of $\mathcal{M}$ reflects the
+adversary's power, i.e. in the first case above the adversary is less powerful
+than the second.
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: Not sure which cases you're talking about here

File: etc/theory/compression_security.tex [hunk 8]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+
+Finally, a noise element is chosen from distribution $\mathcal{V}$. This
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: the distribution

File: etc/theory/compression_security.tex [hunk 9]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+distribution in general describes the uniformly random choice of strings of
```
Comment (dionyziz) [inline etc/theory/compression_security.tex]: The noise distribution may be uniform in our example, but *in general* may not be.

File: etc/theory/compression_security.tex [hunk 10]
```diff
@@ -90,6 +89,138 @@ \section{Introduction}\label{sec:prev}
...
+function} $f$. This function's input is the user's request, resources such as the
+HTML blueprint for the web page, and external functions such as web framework
+middlewares.
+
+The output of the rendering function is the \textit{rendered message} $m$. This
+is the generated HTML response code, that contains response data that is
+rendered by the user's browser and contains different string elements. The HTML
+response elements can be distinguished based on the role of each one in a
+compression-based attack. An element can be a \textit{secret}, \textit{noise},
+\textit{static}, or a \textit{reflection}. An element's alphabet consists of all
```
Comment (dimkarakostas) [inline etc/theory/compression_security.tex]: I mention above that an element is a string, so I prefer the term element here to be more clear in the context of the section.

Comment (dionyziz) [inline etc/theory/compression_security.tex]: ok

**Pull request merged**